applications/crossbar/src/modules/cb_tasks.erl

```diff
@@ -352,20 +391,113 @@ set_db(Context) ->
+-spec read(ne_binary(), cb_context:context(), api_binary(), api_binary()) -> cb_context:context().
 read(TaskId, Context) ->
     AccountId = cb_context:account_id(Context),
+    read(TaskId, set_db(Context), AccountId).
+
+read(TaskId, Context, 'undefined') ->
+    AuthAccountId = cb_context:auth_account_id(Context),
+    read(TaskId, Context, AuthAccountId);
+read(TaskId, Context, AccountId) ->
+    AcceptValue = accept_value(Context),
+    read(TaskId, Context, AccountId, AcceptValue).
+
+-spec accept_value(cb_context:context()) -> api_ne_binary().
+accept_value(Context) ->
+    accept_value(cb_context:req_header(Context, <<"accept">>)
+                ,cb_context:req_value(Context, <<"accept">>)
+                ).
+
+-spec accept_value(api_ne_binary(), api_ne_binary()) -> ne_binary().
+accept_value('undefined', 'undefined') -> ?DEFAULT_CONTENT_TYPE;
+accept_value(Header, 'undefined') -> Header;
+accept_value(_Header, <<"csv">>) -> <<"text/csv">>;
+accept_value(_Header, Tunneled) -> Tunneled.
+
+read(TaskId, Context, AccountId, Accept) ->
+    lager:debug("accept value: ~p", [Accept]),
+    read_doc_or_attachment(TaskId, Context, AccountId, cb_modules_util:parse_media_type(Accept)).
+
+-type parsed_accept_values() :: {'error', 'badarg'} | media_values().
+
+-spec read_doc_or_attachment(ne_binary(), cb_context:context(), ne_binary(), parsed_accept_values()) ->
+                                    cb_context:context().
+read_doc_or_attachment(TaskId, Context, AccountId, {'error', 'badarg'}) ->
+    lager:info("failed to parse the accept header"),
+    read_doc(TaskId, Context, AccountId);
+read_doc_or_attachment(TaskId, Context, AccountId, [?MEDIA_VALUE(<<"application">>, <<"json">>, _, _, _)|_]) ->
+    read_doc(TaskId, Context, AccountId);
+read_doc_or_attachment(TaskId, Context, AccountId, [?MEDIA_VALUE(<<"application">>, <<"x-json">>, _, _, _)|_]) ->
+    read_doc(TaskId, Context, AccountId);
+read_doc_or_attachment(TaskId, Context, AccountId, [?MEDIA_VALUE(<<"*">>, <<"*">>, _, _, _)|_]) ->
+    read_doc(TaskId, Context, AccountId);
+read_doc_or_attachment(TaskId, Context, AccountId, [?MEDIA_VALUE(<<"text">>, <<"csv">>, _, _, _)|_]) ->
+    read_attachment(TaskId, Context, AccountId);
+read_doc_or_attachment(TaskId, Context, AccountId, [?MEDIA_VALUE(<<"text">>, <<"comma-separated-values">>, _, _, _)|_]) ->
+    read_attachment(TaskId, Context, AccountId);
+read_doc_or_attachment(TaskId, Context, AccountId, [?MEDIA_VALUE(<<"application">>, <<"octet-stream">>, _, _, _)|_]) ->
+    read_attachment(TaskId, Context, AccountId);
+read_doc_or_attachment(TaskId, Context, AccountId, [_Accept|Accepts]) ->
+    lager:debug("failed to handle accept value ~p", [_Accept]),
+    read_doc_or_attachment(TaskId, Context, AccountId, Accepts);
+read_doc_or_attachment(TaskId, Context, AccountId, []) ->
+    lager:info("failed to find valid accept value"),
+    read_doc(TaskId, Context, AccountId).
```

Comment:
@jamesaimonetti [your function here](https://github.com/2600hz/kazoo/blob/5bb8459006a859611aa647ec6048c85e27e229c8/applications/crossbar/src/modules/cb_tasks.erl#L425-L443) acts only on its 4th argument. I'm asking that you rewrite it to an arity 1 function, or to something that actually makes sense.